src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -207,6 +207,13 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 
                 PublishCode();
             }
+            catch (Exception)
+            {
+#if READYTORUN
+                PublishEmptyCode();
+#endif
+                throw;
```

Comment:
Rethrowing exceptions makes for an unpleasant debugging experience with first chance exceptions turned on.

All the throws are happening in the dozen or so lines above (starting at line 166): feel free to just `#if !READYTORUN` all the exception throwing above and write your custom logic. E.g. it's unlikely READYTORUN is going to care about the exact exception type being thrown for now (we're not going to generate throwing method bodies, because crossgen doesn't generate them either).

If `exception != IntPtr.Zero || result != CorJitResult.CORJIT_OK`, we can just publish empty code and throw `new Exception()` and be done with it under `READYTORUN`.